cli/src/config.rs

```diff
@@ -958,25 +960,31 @@ mod tests {
                 }
             };
 
-            let (want_existing, want_new) = match self.want {
-                Want::None => (None, None),
-                Want::New(want) => (None, Some(want)),
-                Want::ExistingAndNew(want) => (Some(want), Some(want)),
+            match self.want {
+                Want::None => {}
```

Comment:
Doesn't it remove the assertion for `Want::None` case?

I don't remember the original expectation of these tests, but it doesn't make sense to run a test with no assertion.

Reply:
Thanks for pointing that out, you're exactly right! The `Want::None` test doesn't really make sense, because we need to either call the function that gives us a config path without creating it or with creating it. 

What we want is a way to tell the test to run one or the other, and what to expect out of each. I'm changing it accordingly.

Reply:
It looks like this was added in #1861, so maybe @qfel can clarify?

Reply:
It has been a while, so I don't remember well, but..

It looks like Want::None used to declare both new_config_path and existing_config_path were expected to return None. In this PR test_config_path_home_dir_doesnt_create_new uses Want::None. What do you want to assert for that test? That both existing_config_path and new_or_existing_config_path return None, or something else?

Reply:
I didn't understand the behavior too well before. In this case it should be None with `existing_config_path` and `Some("<path>")` with `new_or_existing_config_path`.